core/src/blocks/block.rs

```diff
@@ -43,6 +43,7 @@ pub struct Env {
     pub state: HashMap<String, Value>,
     pub input: InputState,
     pub map: Option<MapState>,
+    pub secrets: Secrets,
```

Comment:
@spolu pinging you to avoid spending too much time on Rust internal works - is this where I should create a redacting trait? `#[serde(redact)]` type of thing?

Reply:
To be honest I'm not familiar with that trait but looks interesting. What is it?

Reply:
Hm, I thought I needed to create one - because the serialize trait you talked about doesn't seem to be redacting anything, or maybe I just need to read the docs about traits

Reply:
My understanding is that your implementation of the trait (not the default one) would do the redaction?

Reply:
Ok will see how to make a trait implementation then

Reply:
let me know if you need any help here. happy to pair. I don't have the answer off the top of my head but sure we can find an elegant solution here :+1:

Reply:
I got a brief from @flvndvd but I must say this is going to take me a looooooot longer than it would someone with Rust XP, so yeah would definitely love some pair programming there. 🙏

Reply:
GPT-4 has much more Rust experience than all of us put together :P

Reply:
Flav basically told my gpt4 was terrible at Rust ahah

Reply:
This is untrue.

Reply:
GPT-4 is a great resource that can offer some brilliant ideas and hints on Rust programming! However, to really make the most of these suggestions and integrate them into our project, one needs to have some basis around the language. My point was more that we want to ensure that any changes we make not only work smoothly with our existing code but also stay in line with our current coding style. GPT always try to use certain keywords (like 'transpose' and 'unwrap') and non-functional code.

Reply:
It should always always be tried first: https://dust.tt/w/0ec9852c2f/assistant/bb43b15159